treeherder/seta/analyze_failures.py

```diff
@@ -94,14 +110,8 @@ def get_failures_fixed_by_commit():
         # 40char revision we will have two disjunct set of failures
         #
         # Some of this will be improved in https://bugzilla.mozilla.org/show_bug.cgi?id=1323536
-        if revision_id not in failures:
-            failures[revision_id] = []
```

Comment:
With:

failures = defaultdict(list)

You can append items to the list even if the dictionary key doesn't yet exist